### PR TITLE
Add missing type hints in http.client

### PR DESCRIPTION
**Pull request by gruebel**
I'm actually not happy with the type hint for `HTTPConnection.response_class` but changing it to `Type[HttpResponse]` makes `stubtest` complain
```shell
http.client.HTTPConnection.response_class variable differs from runtime type abc.ABCMeta
```
Would be great, if someone could point me into the right direction to better type hint this attribute.

### CODE REVIEW

**Comment by Akuli** [comment]
Use `Type[HTTPResponse]` and add to stubtest's exclude list. At runtime, the type of `HTTPResponse` is `ABCMeta`, because it inherits from `abc.ABCMeta`. But conceptually it should be `Type[HTTPResponse]`.

**Comment by gruebel** [comment]
great, thanks! I was not sure, if I can just add it in the exclude list.

File: stdlib/http/client.pyi [hunk 1]
```diff
@@ -142,6 +145,10 @@ class _HTTPConnectionProtocol(Protocol):
         ) -> HTTPConnection: ...
 
 class HTTPConnection:
+    auto_open: int = ...
```
Comment (bluetech) [inline stdlib/http/client.pyi:148]: I think all of these except for `debuglevel` need to be marked `# undocumented` (see [here](https://github.com/python/typeshed/blob/master/CONTRIBUTING.md#what-to-include)).

Comment (gruebel) [reply stdlib/http/client.pyi]: thanks, added it

**Comment by gruebel** [comment]
@Akuli please have a look again :)

**Review by bluetech — APPROVED** [review]

**Review by Akuli — APPROVED** [review]

**Pull request merged**